presto-docs/src/main/sphinx/admin/cte-materialization.rst

```diff
@@ -0,0 +1,112 @@
+===================
+CTE Materialization
+===================
+
+Common Table Expressions (CTEs) are subqueries that appear in a WITH clause provided by the user.
+Their repeated usage in a query can lead to redundant computations, excessive data retrieval, and high resource consumption.
+
+To address this, Presto supports CTE Materialization allowing intermediate CTEs to be reused within the scope of the same query.
+Materializing CTEs can improve performance when the same CTE is used multiple times in a query by reducing recomputation of the CTE. However, there is also a cost to writing to and reading from disk, so the optimization may not be beneficial for very simple CTEs
+or CTEs that are not used many times in a query.
+Materialized CTEs are stored in temporary tables that are bucketed based on the first projection column.
+To use this feature, the connector used by the query must support the creation of temporary tables. Currently, only the :doc:`/connector/hive` offers this capability.
+
+How to use CTE Materialization
+------------------------------
+
+The following configurations and session properties enable CTE materialization and modify its settings.
+
+``cte-materialization-strategy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Allowed values:** ``ALL``, ``NONE``, ``HEURISTIC``, ``HEURISTIC_COMPLEX_QUERIES_ONLY``
+    * **Default value:** ``NONE``
+
+    Specifies the strategy for materializing Common Table Expressions (CTEs) in queries.
+    ``NONE`` - no CTEs will be materialized.
```

Comment:
These entries - ``NONE``, ``ALL``, ``HEURISTIC``, and ``HEURISTIC_COMPLEX_QUERIES_ONLY`` - look neat in the code here, but when compiled they all pile together in a paragraph. Add a line break before each item to left-justify them.

Screenshot to show the current state. 
<img width="818" alt="Screenshot 2024-05-13 at 2 32 32 PM copy" src="https://github.com/prestodb/presto/assets/7013443/1a68b118-233d-4657-9a70-82078d9ebb0b">